researchstack-sdk/src/main/java/org/sagebionetworks/bridge/researchstack/BridgeDataProvider.java

```diff
@@ -18,569 +16,503 @@
+
+    private final StorageAccessWrapper storageAccessWrapper;
+    private final ResearchStackDAO researchStackDAO;
+    private final BridgeManagerProvider bridgeManagerProvider;
+    private final BridgeConfig bridgeConfig;
+    private final AuthenticationManager authenticationManager;
+    private final ConsentManager consentManager;
+
+
+    //used by tests to mock service
+    BridgeDataProvider(ResearchStackDAO researchStackDAO, StorageAccessWrapper storageAccessWrapper,
+                       TaskHelper taskHelper, UploadHandler uploadHandler) {
+        this.researchStackDAO = researchStackDAO;
+        this.storageAccessWrapper = storageAccessWrapper;
+        this.taskHelper = taskHelper;
+        this.uploadHandler = uploadHandler;
+
+        this.bridgeManagerProvider = BridgeManagerProvider.getInstance();
+
+        // convenience accessors
+        this.bridgeConfig = bridgeManagerProvider.getBridgeConfig();
+        this.authenticationManager = bridgeManagerProvider.getAuthenticationManager();
+        this.consentManager = bridgeManagerProvider.getConsentManager();
+
+    }
+
+    public BridgeDataProvider(BridgeManagerProvider bridgeManagerProvider,
+                              ResourcePathManager.Resource publicKey) {
+        this.researchStackDAO = new ResearchStackDAO(bridgeManagerProvider.getApplicationContext());
+        this.bridgeManagerProvider = bridgeManagerProvider;
+        // convenience accessors
+        this.bridgeConfig = bridgeManagerProvider.getBridgeConfig();
+        this.authenticationManager = bridgeManagerProvider.getAuthenticationManager();
+        this.consentManager = bridgeManagerProvider.getConsentManager();
+
+        this.storageAccessWrapper = new StorageAccessWrapper();
+        this.uploadHandler = new UploadHandler(bridgeManagerProvider.getApplicationContext(),
+                storageAccessWrapper, publicKey);
+        this.taskHelper = new TaskHelper(storageAccessWrapper, ResourceManager.getInstance(),
+                AppPrefs.getInstance(), uploadHandler);
+    }
+
+
+    @Override
+    public Observable<DataResponse> initialize(Context context) {
+        logger.debug("Called initialize");
+
+        return SUCCESS_DATA_RESPONSE;
+    }
+
+    @Override
+    public String getStudyId() {
+        return bridgeConfig.getStudyId();
+    }
+
+    //region Consent
+
+    @Override
+    public Observable<DataResponse> withdrawConsent(Context context, String reason) {
+        logger.debug("Called withdrawConsent");
+        //TODO: allow withdrawal from specific subpopulation
+
+        return withdrawAllConsents(reason).andThen(SUCCESS_DATA_RESPONSE);
+    }
+
+    @NonNull
+    public Completable withdrawConsent(@NonNull String subpopulationGuid, @Nullable String reason) {
+        return consentManager.withdrawConsent(subpopulationGuid, reason);
+    }
+
+
+    @NonNull
+    public Completable withdrawAllConsents(@Nullable String reason) {
```

Comment:
Are these methods intended for external use?  It seems as though it is just an internal helper to manage the return type differences between DataProvider and the managers?  Should it be public?